scripts/data/generate_utreexo_data.py

```diff
@@ -47,20 +45,34 @@ def serialize(self):
             pending_word = 0
             pending_word_len = 0
 
-        data.append(len(sub_data))
-        data.extend(sub_data)
-
         # Check if there's still a word pending
         if pending_word_len > 0:
-            data.append(pending_word)
-        data.append(pending_word_len)
+            sub_data.append(pending_word)
+
+        sub_data.append(pending_word_len)
+
+        return sub_data
+
+    def serialize(self) -> OrderedDict:
```

Comment:
We need serialize for hashing only, so i'd keep only one flow (flexibility is good though!)